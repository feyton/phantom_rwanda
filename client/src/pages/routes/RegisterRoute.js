import React from 'react';
import { toast } from 'react-toastify';
import { axiosBase as axios } from '../../utils/Api.js';
import RouteComponent from './components/RouteComponent.js';

function RegisterRoute() {
  const registerRouteToDB = async (data) => {
    try {
      await axios.post('/routes', data);
      toast('Route Created successfully', { type: 'success' });
      return;
    } catch (error) {
      return Promise.reject(error);
    }
  };
  return (
    <RouteComponent formAction={registerRouteToDB} formTitle="Register Route" />
  );
}

export default RegisterRoute;
