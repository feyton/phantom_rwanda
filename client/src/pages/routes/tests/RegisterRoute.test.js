import { mount } from 'enzyme';
import { enableFetchMocks } from 'jest-fetch-mock';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import RegisterRoute from '../RegisterRoute.js';

enableFetchMocks();
describe('Register Page', () => {
  const navigateMock = () => {
    return jest.fn();
  };

  const mockedUsedNavigate = jest.fn();
  jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUsedNavigate
  }));
  const mockedUsedLocation = jest.fn();
  jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useLocation: () => mockedUsedLocation
  }));
  it('should render the register component', () => {
    const elem = renderer
      .create(
        <MemoryRouter>
          <RegisterRoute />
        </MemoryRouter>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('should render the Route component', () => {
    const wrapper = mount(
      <MemoryRouter>
        <RegisterRoute />
      </MemoryRouter>
    );
    expect(wrapper.find('RouteComponent').length).toEqual(1);
  });
  it('should Post data from In the database', async () => {
    const route = {
      destination1: 'Kabuga',
      destination2: 'Nyabugogo',
      distance: '20km'
    };
    global.fetch.mockResponseOnce(
      JSON.stringify({
        routes: {
          destination1: 'Kabuga',
          destination2: 'Nyabugogo',
          distance: '20km'
        }
      })
    );

    expect(global.fetch.mock.calls.length).toEqual(0);
  });
  it('It should test the registerRouteToDB function', () => {
    const registerRouteToDB = jest.fn();
    const component = mount(
      <MemoryRouter>
        <RegisterRoute registerRouteToDB={registerRouteToDB()} />
      </MemoryRouter>
    );
    const form = component.find('form');
    form.simulate('submit');
    expect(registerRouteToDB).toBeCalledTimes(1);
  });
});
