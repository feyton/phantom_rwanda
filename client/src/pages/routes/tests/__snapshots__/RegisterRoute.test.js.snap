// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Page should render the register component 1`] = `
<div
  className="w-full px-5"
>
  <form
    className="p-8 shadow-main px-4 flex flex-col w-100 mx-auto"
    onSubmit={[Function]}
  >
    <h1
      className="font-bold text-xl pb-8 text-center font-raleway"
    >
       
      Register Route
       
    </h1>
    <div
      className="block mb-4 xl:flex xl:justify-center md:grid md:grid-cols-2 md:gap-2"
    >
      <div
        className=""
      >
        <label
          className="block text-grey-darker text-sm font-bold mb-2"
          htmlFor="origin"
        >
          Origin:
        </label>
        <input
          className="appearance-none border font-raleway rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
          name="origin"
          onChange={[Function]}
          placeholder="Ex: Nyabugogo"
          required={true}
        />
      </div>
      <div>
        <label
          className="block text-grey-darker text-sm font-bold mb-2"
          htmlFor="destination"
          name="distance"
        >
          Destination
        </label>
        <input
          className="appearance-none border font-raleway rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
          name="destination"
          onChange={[Function]}
          placeholder="Ex: Down-Town"
          required={true}
          type="text"
        />
      </div>
    </div>
    <div
      className="h-100 w-100 p-2 m-3 "
    >
      <div
        className="h-screen md:h-[60vh] w-[100%]"
      />
    </div>
    <div
      className="self-center mt-6"
    >
      <div>
        <button
          className="bg-primary text-sm text-white py-2 px-3 hover:bg-hover rounded-xl py-2 text-base md:text-sm px-5"
          type="submit"
        >
          Register Route
        </button>
      </div>
    </div>
  </form>
</div>
`;
